src/main/java/org/alfresco/rest/api/search/impl/ResultMapper.java

```diff
@@ -635,4 +642,31 @@ protected SolrJSONResultSet findSolrResultSet(ResultSet results)
 
         return null;
     }
+    public CollectionWithPagingInfo<TupleList> toCollectionWithPagingInfo(JSONArray docs, SearchSQLQuery searchQuery) throws JSONException
+    {
+        if(docs == null)
+        {
+            throw new RuntimeException("Solr response is required instead of docs:" + docs);
```

Comment:
Can we change the error message here? docs is null here. The error message does not make much sense to me.